### ISSUE TITLE
Panic because of invalid syntax (literal instead of ident)

### ISSUE THREAD

**Issue by ark0f** [rust-lang/rustfmt#3418]
Code:
```rust
struct Message {
    entities: Vec<"MessageEntity">,
}
```

<details>
    <summary>Panic message</summary>

```
thread 'main' panicked at 'internal error: entered unreachable code', src\tools\rustfmt\src\types.rs:155:43
stack backtrace:
   0: std::sys_common::backtrace::_print
             at src\libstd\sys\windows\backtrace/mod.rs:94
             at src\libstd\sys\windows\backtrace/mod.rs:81
             at src\libstd\sys_common/backtrace.rs:70
   1: std::panicking::default_hook::{{closure}}
             at src\libstd\sys_common/backtrace.rs:58
             at src\libstd/panicking.rs:200
   2: std::panicking::default_hook
             at src\libstd/panicking.rs:215
   3: std::panicking::rust_panic_with_hook
             at src\libstd/panicking.rs:478
   4: std::panicking::begin_panic
   5: <core::iter::adapters::chain::Chain<A, B> as core::iter::traits::iterator::Iterator>::fold
   6: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T, I>>::from_iter
   7: rustfmt_nightly::types::rewrite_segment
   8: rustfmt_nightly::types::rewrite_path
   9: rustfmt_nightly::types::<impl rustfmt_nightly::rewrite::Rewrite for syntax::ast::Ty>::rewrite
  10: rustfmt_nightly::items::rewrite_struct_field
  11: <rustfmt_nightly::lists::ListItems<'a, I, F1, F2, F3> as core::iter::traits::iterator::Iterator>::next
  12: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T, I>>::from_iter
  13: rustfmt_nightly::vertical::rewrite_with_alignment
  14: rustfmt_nightly::items::format_struct
  15: rustfmt_nightly::visitor::FmtVisitor::visit_item
  16: rustfmt_nightly::reorder::<impl rustfmt_nightly::visitor::FmtVisitor<'a>>::visit_items_with_reordering
  17: rustfmt_nightly::visitor::FmtVisitor::walk_mod_items
  18: rustfmt_nightly::visitor::FmtVisitor::format_separate_mod
  19: rustfmt_nightly::formatting::format_project
  20: syntax::with_globals
  21: <rustfmt_nightly::Session<'b, T>>::format
  22: rustfmt::format_and_emit_report
  23: <rustfmt_nightly::Session<'b, T>>::override_config
  24: rustfmt::execute
  25: rustfmt::main
  26: std::rt::lang_start::{{closure}}
  27: std::panicking::try::do_call
             at src\libstd/rt.rs:49
             at src\libstd/panicking.rs:297
  28: _rust_maybe_catch_panic
             at src\libpanic_unwind/lib.rs:87
  29: std::rt::lang_start_internal
             at src\libstd/panicking.rs:276
             at src\libstd/panic.rs:388
             at src\libstd/rt.rs:48
  30: main
  31: _tmainCRTStartup
  32: mainCRTStartup
  33: unit_addrs_search
  34: unit_addrs_search

**Comment by rchaser53** [comment]
this issue will be resolved in #3336

**Comment by rchaser53** [comment]
Sorry. Your code is a just syntax error, not related with #3336.
It should emit the error message like below.

```
error: expected type, found `"MessageEntity"`
 --> tests/source/const_generics.rs:2:14
  |
2 |     field1: <"MessageEntity">,
  |              ^^^^^^^^^^^^^^^
```

rustfmt will emit above the error when rustfmt update rustc parser. Please wait.
